### PR TITLE
USER-INTEL: enforce icc and add icc flags

### PR DESCRIPTION
**Pull request by junghans**
ToDo:
- [x]  Add user overwrite
- [x] Check if compiler flags actually work

### CODE REVIEW

**Comment by andeplane** [comment]
Does USER-INTEL require intel compiler? One of my students compiled with this package using gcc and might enjoy the extra performance because of the data structure in the force loop even with gcc. Am I mistaken here maybe?

**Comment by akohlmey** [comment]
@andeplane you can compile USER-INTEL without the intel compilers, but it is so full of directives and code constructs, that only the intel compilers can make good use of, it is not worth for "normal people" to use this package without. performance will be subpar. we try to keep it compilable with gcc, so that those files can be included in our continuous integration checking.

**Comment by wmbrownIntel** [comment]
I don’t think that we need a formal restriction for this. It should compile without, but I don’t think one would want to in most cases. There are some things like neighbor padding (done for processors before Skylake/KNL) that will slow down the code if it is not vectorized. But I haven’t really evaluated performance with GNU lately.

At some point in the near future, the vectorization directives will all be converted to use the OpenMP standard. This should help a lot for use with other compilers, I hope.

Best, - Mike

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -603,6 +603,18 @@ if(PKG_OPT)
...
+    endif()
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "2016")
+      message(FATAL_ERROR "USER-INTEL is needed at least 2016 intel compiler")
+    endif()
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "2017.3")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -xHost")
+    else()
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -xCOMMON-AVX512")
+    endif()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -qopenmp -qno-offload -fno-alias -ansi-alias -restrict -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG-O2 -fp-model fast=2 -no-prec-div -qoverride-limits -qopt-zmm-usage=high")
```
Comment (junghans) [inline cmake/CMakeLists.txt]: @wmbrownIntel : are these the right flags `-qopenmp -qno-offload -fno-alias -ansi-alias -restrict -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG-O2 -fp-model fast=2 -no-prec-div -qoverride-limits -qopt-zmm-usage=high` plus `-xHost` (or -`xCOMMON-AVX512` for >-icc-2017.3)?

**Comment by wmbrownIntel** [comment]
> are these the right flags -qopenmp -qno-offload -fno-alias -ansi-alias -restrict -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG-O2 -fp-model fast=2 -no-prec-div -qoverride-limits -qopt-zmm-usage=high plus -xHost (or -xCOMMON-AVX512 for >-icc-2017.3)?
There should be a space between RNG-O2 (RNG -O2) and -xCOMMON-AVX512 isn’t necessary for 2018 and above, just the two 2017 oddballs ☺.

**Comment by junghans** [comment]
@sjplimp  works for me! @akohlmey please test!

**Pull request merged**